cns/hnsclient/hnsclient_windows.go

```diff
@@ -604,6 +608,17 @@ func CreateHostNCApipaEndpoint(
 	return endpoint.Id, nil
 }
 
+// adjustIPConfig applies change on gw IP address.
+// Currently, cns using the same ip address "169.254.128.1" for both apipa gw and loopback adapter. This cause conflict issue when hns get restarted and not able to rehydrate the apipa endpoints.
+// This func is to overwrite the address to 169.254.128.2 when the gateway address is 169.254.128.1
+func adjustIPConfig(localIPConfiguration *cns.IPConfiguration) {
```

Comment:
let's rename the func to indicate which IP config is getting updated? something along the lines of 
UpdateGwForLocalIPConfiguration